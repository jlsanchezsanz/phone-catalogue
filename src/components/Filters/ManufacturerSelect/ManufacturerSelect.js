import React from 'react';
import PropTypes from 'prop-types';

import { MANUFACTURERS } from '../../../constants/manufacturers';

export default function ManufacturerSelect({ onManufacturerChange }) {
  function handleManufacturerChange(e) {
    const { value, name } = e.target;
    onManufacturerChange({ [name]: value });
  }

  return (
    <select
      name='manufacturer'
      className='form-control'
      onChange={handleManufacturerChange}
    >
      {Object.keys(MANUFACTURERS).map((manufacturer) => (
        <option key={manufacturer} value={MANUFACTURERS[manufacturer]}>
          {manufacturer}
        </option>
      ))}
    </select>
  );
}

ManufacturerSelect.propTypes = {
  onManufacturerChange: PropTypes.func.isRequired
};
