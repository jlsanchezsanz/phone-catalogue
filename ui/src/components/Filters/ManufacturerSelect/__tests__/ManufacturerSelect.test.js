import React from 'react';
import { shallow } from 'enzyme';

import ManufacturerSelect from '../ManufacturerSelect';

const setUpShallow = (handleManufacturerChange) => {
  const component = shallow(
    <ManufacturerSelect onManufacturerChange={handleManufacturerChange} />
  );
  return component;
};

describe('ManufacturerSelect', () => {
  const handleManufacturerChange = jest.fn();

  it('should render component', () => {
    const component = setUpShallow(handleManufacturerChange);
    expect(component).toMatchSnapshot();
  });

  it('should execute callback on manufacturer change', () => {
    const name = 'manufacturer';
    const value = 'Nokia';
    const component = setUpShallow(handleManufacturerChange);
    const select = component.find(`select[name="${name}"]`);

    expect(handleManufacturerChange).not.toHaveBeenCalled();
    select.simulate('change', { target: { value, name } });

    expect(handleManufacturerChange).toHaveBeenCalledTimes(1);
    expect(handleManufacturerChange).toHaveBeenCalledWith({
      [name]: value
    });
  });
});
