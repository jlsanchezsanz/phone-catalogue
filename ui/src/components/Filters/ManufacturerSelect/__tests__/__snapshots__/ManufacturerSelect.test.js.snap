// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManufacturerSelect should render component 1`] = `
<select
  className="form-control"
  name="manufacturer"
  onChange={[Function]}
>
  <option
    key="All Manufacturers"
    value=""
  >
    All Manufacturers
  </option>
  <option
    key="Apple"
    value="Apple"
  >
    Apple
  </option>
  <option
    key="Asus"
    value="Asus"
  >
    Asus
  </option>
  <option
    key="Huawei"
    value="Huawei"
  >
    Huawei
  </option>
  <option
    key="Motorola"
    value="Motorola"
  >
    Motorola
  </option>
  <option
    key="Nokia"
    value="Nokia"
  >
    Nokia
  </option>
  <option
    key="Samsung"
    value="Samsung"
  >
    Samsung
  </option>
  <option
    key="Xiaom"
    value="Xiaomi"
  >
    Xiaom
  </option>
</select>
`;
